analysis/analysis.go

```diff
@@ -175,9 +180,17 @@ func (c *AnalysisController) runMeasurements(run *v1alpha1.AnalysisRun, tasks []
 				newMeasurement.Status = v1alpha1.AnalysisStatusError
 			} else {
 				if t.incompleteMeasurement == nil {
-					newMeasurement, err = provider.Run(t.metric, run.Spec.Arguments)
+					newMeasurement, err = provider.Run(run, t.metric, run.Spec.Arguments)
 				} else {
-					newMeasurement, err = provider.Resume(t.metric, run.Spec.Arguments, *t.incompleteMeasurement)
+					// metric is incomplete. either terminate or resume it
+					if terminating {
+						newMeasurement, err = provider.Terminate(run, t.metric, run.Spec.Arguments, *t.incompleteMeasurement)
+						if err == nil && newMeasurement.Status == v1alpha1.AnalysisStatusSuccessful {
```

Comment:
It feels weird to call a terminated metric succesful, but Idk what else I would call it

Reply:
Also, while this section doesn't have too complex code, I think we should probably have some unit tests here since this is one of the key sections of the analysis runs.